File: src/main/java/no/fintlabs/model/statistics/IntegrationStatisticsFilter.java
package no.fintlabs.model.statistics;

import lombok.Builder;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

import java.util.Collection;

@Getter
@Jacksonized
@Builder(toBuilder = true)
public class IntegrationStatisticsFilter {
    private final Collection<Long> sourceApplicationIds;
    private final Collection<String> sourceApplicationIntegrationIds;
    private final Collection<Long> integrationIds;
}

File: src/main/kotlin/no/novari/flyt/history/model/statistics/IntegrationStatisticsFilter.kt
package no.novari.flyt.history.model.statistics

data class IntegrationStatisticsFilter(
    val sourceApplicationIds: Collection<Long>? = null,
    val sourceApplicationIntegrationIds: Collection<String>? = null,
    val integrationIds: Collection<Long>? = null,
) {
    companion object {
        @JvmStatic
        fun builder() = Builder()
    }

    class Builder {
        private var sourceApplicationIds: Collection<Long>? = null
        private var sourceApplicationIntegrationIds: Collection<String>? = null
        private var integrationIds: Collection<Long>? = null

        fun sourceApplicationIds(sourceApplicationIds: Collection<Long>?) =
            apply {
                this.sourceApplicationIds = sourceApplicationIds
            }

        fun sourceApplicationIntegrationIds(sourceApplicationIntegrationIds: Collection<String>?) =
            apply {
                this.sourceApplicationIntegrationIds = sourceApplicationIntegrationIds
            }

        fun integrationIds(integrationIds: Collection<Long>?) =
            apply {
                this.integrationIds = integrationIds
            }

        fun build() =
            IntegrationStatisticsFilter(
                sourceApplicationIds = sourceApplicationIds,
                sourceApplicationIntegrationIds = sourceApplicationIntegrationIds,
                integrationIds = integrationIds,
            )
    }
}
